gitpod-io/gitpod
# [tests] run Workspace Integration tests in Github Actions

## Description

Run Workspace integration tests in GHA. _Note: it is not a reliable source of truth yet  - we're iterating._

Many thanks to @vulkoingim for getting us online with GHA, and @iQQBot for pioneering with the IDE integration tests.

## Related Issue(s)

Fixes # n/a

## How to test

1. Use the `gh` CLI to trigger a run yourself, you'll need to authenticate `gh auth login`, and then trigger the action `gh workflow run "Workspace integration tests" --ref kylos101/wks-intg-fix -f name=kylos101-w723fcbd2de -f version=main-gha.4902`
2. Inspect GHA versus werft results, below

[GHA](https://github.com/gitpod-io/gitpod/actions/runs/4449392181), runs tests, there are failures and the # of tests run varies (to fix in a future PR), status is shared in Slack via #workspace-jobs.

compare it with

[The Werft job](https://werft.gitpod-dev.com/job/gitpod-custom-main.642), which is broke as of ~March 15, and no longer can run tests because it cannot spin up a preview environment, [without us investing more on the SID team](https://gitpod.slack.com/archives/C032A46PWR0/p1679071081010059?thread_ts=1679000508.802599&cid=C032A46PWR0). @corneliusludmann for :eyes: 

## Release Notes

```release-note
NONE
```

## Documentation


## Build Options:

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
